drivers/hcsr04/hcsr04.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2021 AUTH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_hcsr04
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the HC-SR04
+ *
+ * @author      Evripidis Chondromatidis <eurichon1996@gmail.com>
+ *
+ * @}
+ */
+
+#include "hcsr04.h"
+#include "hcsr04_constants.h"
+#include "hcsr04_params.h"
+
+#include "periph/gpio.h"
+#include "xtimer.h"
+
+#include <stdint.h>
+#include <errno.h>
+#include <math.h>
+
+
+void hcsr04_int_callback(void *arg){
+    uint32_t now = xtimer_now_usec();
+    hcsr04_t *p_dev = (hcsr04_t *)arg;
+
+    if (gpio_read(p_dev->params.echo_pin)) {
+        p_dev->pre_meas_t_us = now;
+    } else {
+        uint32_t dt_us_one_way = (now - p_dev->pre_meas_t_us) / 2;
+        p_dev->distance = 1e-3 * p_dev->sound_speed * dt_us_one_way;
+    }
+}
+
+int hcsr04_init(hcsr04_t *dev, const hcsr04_params_t *params)
+{
+    if (!dev || ! params) {
+        return -EINVAL;
+    }
```

Comment:
```suggestion
    assert(dev && params);
```

Commonly most RIOT drivers use assertions to check for parameters 😉